lib/rabbitmq_message_deduplication/cache.ex

```diff
@@ -142,6 +143,20 @@ defmodule RabbitMQMessageDeduplication.Cache do
     end
   end
 
+  def list_caches() do
+      Mnesia.foldl(fn(rec, acc) -> [rec | acc] end, [], @caches)
+  end
+
+  @doc """
+  All cache tables should be distributed if possible
+  """
+  @spec ensure_distributed()
+  def ensure_distributed() do
+    cache_tables = list_caches()
+    Enum.each(cache_tables, _ensure_distributed)
```

Comment:
You need to pass a reference to the function `maybe_distribute_cache` using `&` and you need to specify the arity of the function as well.

```elixir
Enum.each(cache_tables, &maybe_distribute_cache/1)
```

Nevertheless, see comment above.